assets/js/components/KeyMetrics/ConnectGA4CTAWidget.test.js

```diff
@@ -0,0 +1,215 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import ConnectGA4CTAWidget from './ConnectGA4CTAWidget';
+import {
+	CORE_USER,
+	KM_ANALYTICS_ENGAGED_TRAFFIC_SOURCE,
+	KM_ANALYTICS_LOYAL_VISITORS,
+	KM_ANALYTICS_NEW_VISITORS,
+	KM_ANALYTICS_TOP_TRAFFIC_SOURCE,
+	KM_SEARCH_CONSOLE_POPULAR_KEYWORDS,
+	KM_ANALYTICS_ADSENSE_TOP_EARNING_CONTENT,
+} from '../../googlesitekit/datastore/user/constants';
+import {
+	render,
+	createTestRegistry,
+	provideModules,
+	provideKeyMetrics,
+	provideUserAuthentication,
+	provideUserCapabilities,
+} from '../../../../tests/js/test-utils';
+import { CORE_WIDGETS } from '../../googlesitekit/widgets/datastore/constants';
+import { AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY } from '../../googlesitekit/widgets/default-areas';
+import { CONTEXT_MAIN_DASHBOARD_KEY_METRICS } from '../../googlesitekit/widgets/default-contexts';
+
+describe( 'ConnectGA4CTAWidget', () => {
+	let registry;
+
+	const DISMISSED_ITEM_KEY = 'key-metrics-connect-ga4-cta-widget';
+
+	const Widget = ( { children } ) => <div>{ children }</div>;
+	const WidgetNull = () => <div>NULL</div>;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+
+		provideUserAuthentication( registry );
+		provideUserCapabilities( registry );
+		provideKeyMetrics( registry );
+
+		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
+		registry.dispatch( CORE_USER ).receiveIsUserInputCompleted( true );
+
+		provideModules( registry, [
+			{
+				slug: 'analytics-4',
+				active: false,
+				connected: false,
+			},
+		] );
+	} );
+
+	it( 'should not render when the widget is dismissed', async () => {
+		registry
+			.dispatch( CORE_USER )
+			.receiveGetDismissedItems( [ DISMISSED_ITEM_KEY ] );
+
+		const { container, waitForRegistry } = render(
+			<ConnectGA4CTAWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
+			{
+				registry,
+			}
+		);
+		await waitForRegistry();
+		expect( container ).toHaveTextContent( 'NULL' );
+	} );
+
+	it( 'should not render if user input is not completed', async () => {
+		global._googlesitekitUserData.isUserInputCompleted = false;
+		await registry
+			.dispatch( CORE_USER )
+			.receiveIsUserInputCompleted( false );
+
+		const { container, waitForRegistry } = render(
+			<ConnectGA4CTAWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
+			{
+				registry,
+			}
+		);
+		await waitForRegistry();
+		expect( container ).toHaveTextContent( 'NULL' );
+	} );
+
+	it( 'should not render if GA4 is connected', async () => {
+		provideModules( registry, [
+			{
+				slug: 'analytics-4',
+				active: true,
+				connected: true,
+			},
+		] );
+
+		const { container, waitForRegistry } = render(
+			<ConnectGA4CTAWidget Widget={ Widget } WidgetNull={ WidgetNull } />,
+			{
+				registry,
+			}
+		);
+		await waitForRegistry();
+		expect( container ).toHaveTextContent( 'NULL' );
+	} );
```

Comment:
This should probably also be another condition where we guard the widget from rendering from outside.